diff --git a/src/na/na_ofi.c b/src/na/na_ofi.c
index 379f1eb..d883d0b 100644
--- a/src/na/na_ofi.c
+++ b/src/na/na_ofi.c
@@ -278,16 +278,14 @@ struct na_ofi_op_id {
 static NA_INLINE void
 na_ofi_domain_lock(struct na_ofi_domain *domain)
 {
-    if (domain->nod_prov_type == NA_OFI_PROV_VERBS ||
-        domain->nod_prov_type == NA_OFI_PROV_PSM2)
+    if (domain->nod_prov_type == NA_OFI_PROV_PSM2)
         hg_thread_mutex_lock(&domain->nod_mutex);
 }
 
 static NA_INLINE void
 na_ofi_domain_unlock(struct na_ofi_domain *domain)
 {
-    if (domain->nod_prov_type == NA_OFI_PROV_VERBS ||
-        domain->nod_prov_type == NA_OFI_PROV_PSM2)
+    if (domain->nod_prov_type == NA_OFI_PROV_PSM2)
         hg_thread_mutex_unlock(&domain->nod_mutex);
 }
 
@@ -297,8 +295,6 @@ na_ofi_class_lock(na_class_t *na_class)
     struct na_ofi_private_data *priv = NA_OFI_PRIVATE_DATA(na_class);
     struct na_ofi_domain *domain = priv->nop_domain;
 
-    if (domain->nod_prov_type == NA_OFI_PROV_VERBS)
-        hg_thread_mutex_lock(&priv->nop_mutex);
 }
 
 static NA_INLINE void
@@ -306,9 +302,6 @@ na_ofi_class_unlock(na_class_t *na_class)
 {
     struct na_ofi_private_data *priv = NA_OFI_PRIVATE_DATA(na_class);
     struct na_ofi_domain *domain = priv->nop_domain;
-
-    if (domain->nod_prov_type == NA_OFI_PROV_VERBS)
-        hg_thread_mutex_unlock(&priv->nop_mutex);
 }
 
 static NA_INLINE na_bool_t
